CMakeLists.txt

```diff
@@ -355,6 +355,14 @@ if(NOT EXISTS ${CMAKE_BINARY_DIR}/.config OR NOT "${NUTTX_DEFCONFIG}" STREQUAL
       ${NUTTX_DEFCONFIG}
       CACHE INTERNAL "Saved defconfig path" FORCE)
 
+  # update value of CONFIG_BASE_DEFCONFIG
+
+  execute_process(
+    COMMAND
+      sed -ie
```

Comment:
Could you please use cmake-related methods to extract configurations? Currently cmake also supports the Visual Studio msvc platform https://github.com/apache/nuttx/pull/6897), It is better to remove the coupling of host operating system.

Reply:
Do you have any hints? I am unfamilar with them yet. I saw someone uses file(READ) string(REPLACE) then file(WRITE). are there better ways?

Reply:
defconfig has changed after cmake configuration, try `cmake --build build -t savedefconfig` to reproduce this issue.

```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   boards/sim/sim/sim/configs/ostest/defconfig
```

@xuxin930  could you please take a look this issue? Is there a better way to use cmake to implement the feature of BASE_DEFCONFIG?

Reply:
@anchao thanks for pinpointing the cause of CI failure. I pushed an update which works locally.

Interestingly it seems that our CMake system has already been using tools like `cat`, `sed`, `grep`, `sort`, `uniq` etc?

On the other side, my attempt of `string(REGEX REPLACE)` here wasn'ts successful and the CMake native approach looks so tedious compared with the `sed` approach. 

Let's see how this update goes and if @xuxin930 has a better way.

Reply:
> @xuxin930 could you please take a look this issue? Is there a better way to use cmake to implement the feature of BASE_DEFCONFIG?

hi @anchao @yf13 
Since CMake does not provide a cross-platform text processing method, can we workaround this case like this?
1. keep `CONFIG_BASE_DEFCONFIG` within the build system `ENV`
2. mark the **_dirty status_** of BASE_DEFCONFIG following `.config`
3. manually write `config.h` with `nuttx_mkconfig.cmake`

I think this is what this config requires.
 Even if we reset a config to DEFCOFING during CMake configuration, it will still complete these things.
***
In addition, I noticed that this patch does not monitor the BASE_DEFCONFIG status.
If you must modify the configuration file, you can try `file(READ)-string(REPLACE)-file(WRITE)` above. 
Or try to achieve a `kconfiglib` independent `setconfig` target?

Reply:
@xuxin930 thanks!
- did you mean that we can modify `nuttx_mkconfig` instead via a `file(WRITE "#define ...")` operation?  
- Is dirty status monitoring done by comparing `.config` with `.config.orig`? if they differ, then add "-dirty" to the string?